src/panels/lovelace/editor/config-elements/hui-entities-editor.ts

```diff
@@ -0,0 +1,318 @@
+import { mdiDelete, mdiDrag, mdiListBox, mdiPencil, mdiPlus } from "@mdi/js";
+import { ComboBoxLightOpenedChangedEvent } from "@vaadin/combo-box/vaadin-combo-box-light";
+import { CSSResultGroup, LitElement, css, html, nothing } from "lit";
+import { customElement, property, query, state } from "lit/decorators";
+import { repeat } from "lit/directives/repeat";
+import { fireEvent } from "../../../../common/dom/fire_event";
+import { preventDefault } from "../../../../common/dom/prevent_default";
+import { stopPropagation } from "../../../../common/dom/stop_propagation";
+import "../../../../components/entity/ha-entity-picker";
+import type { HaEntityPicker } from "../../../../components/entity/ha-entity-picker";
+import "../../../../components/ha-button";
+import "../../../../components/ha-icon-button";
+import "../../../../components/ha-list-item";
+import "../../../../components/ha-sortable";
+import "../../../../components/ha-svg-icon";
+import { HomeAssistant } from "../../../../types";
+
+type EntityConfig = {
+  entity: string;
+};
+
+declare global {
+  interface HASSDomEvents {
+    "edit-entity": { index: number };
+  }
+}
+
+@customElement("hui-entities-editor")
+export class HuiEntitiesEditor extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property({ attribute: false })
+  public entities?: EntityConfig[];
+
+  @query(".add-container", true) private _addContainer?: HTMLDivElement;
+
+  @query("ha-entity-picker") private _entityPicker?: HaEntityPicker;
+
+  @state() private _addMode = false;
+
+  private _opened = false;
+
+  private _entitiesKeys = new WeakMap<EntityConfig, string>();
+
+  private _getKey(entity: EntityConfig) {
+    if (!this._entitiesKeys.has(entity)) {
+      this._entitiesKeys.set(entity, Math.random().toString());
+    }
+
+    return this._entitiesKeys.get(entity)!;
+  }
+
+  protected render() {
+    if (!this.entities || !this.hass) {
+      return nothing;
+    }
+
+    return html`
+      <ha-expansion-panel outlined>
```

Comment:
Not sure the expansion panel should be part of this element, if we want to reuse this in the entities card editor for example it would not be an expansion panel I think?

Reply:
It was like this for card features but I agree, we can extract it and keep the expansion-panel outside.